easybuild/tools/filetools.py

```diff
@@ -763,6 +763,21 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    # Attempt extracting the patch if it ends in .patch.gz, .patch.bz2, .patch.xz
+    # split in name + extension
+    apatch_root, apatch_file = os.path.split(apatch)
+    apatch_name, apatch_extension = os.path.splitext(apatch_file)
+    if apatch_extension in ['.gz','.bz2','.xz']:
```

Comment:
I did not add zip because zip can be an archive. I'm only supporting ".patch.<compression extension>", which are not archive. 

I don't think we can make the script clever enough to support archive of multiple patches.